pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateData.java

```diff
@@ -37,19 +36,23 @@ public record ServiceUnitStateData(
         }
     }
 
-    public ServiceUnitStateData(ServiceUnitState state, String broker, String sourceBroker) {
-        this(state, broker, sourceBroker, false, System.currentTimeMillis());
+    public ServiceUnitStateData(ServiceUnitState state, String broker, String sourceBroker, long versionId) {
+        this(state, broker, sourceBroker, false, System.currentTimeMillis(), versionId);
     }
 
-    public ServiceUnitStateData(ServiceUnitState state, String broker) {
-        this(state, broker, null, false, System.currentTimeMillis());
+    public ServiceUnitStateData(ServiceUnitState state, String broker, long versionId) {
+        this(state, broker, null, false, System.currentTimeMillis(), versionId);
     }
 
-    public ServiceUnitStateData(ServiceUnitState state, String broker, boolean force) {
-        this(state, broker, null, force, System.currentTimeMillis());
+    public ServiceUnitStateData(ServiceUnitState state, String broker, boolean force, long versionId) {
+        this(state, broker, null, force, System.currentTimeMillis(), versionId);
     }
 
     public static ServiceUnitState state(ServiceUnitStateData data) {
         return data == null ? ServiceUnitState.Init : data.state();
     }
+
+    public static long versionId(ServiceUnitStateData data) {
+        return data == null ? 0 : data.versionId();
```

Comment:
VERSION_ID_INIT is 1, and for the null data, it needs to be zero in order to make the first version to be 1( 0 + 1).

We can return VERSION_ID_INIT - 1(which is zero) if null.

Reply:
Maybe we can modify the method `getNextVersionId(ServiceUnitStateData data)` like this and reuse it in method `getNextVersionId(String serviceUnit)`.

```
    private long getNextVersionId(ServiceUnitStateData data) {
        if (data == null) {
            return VERSION_ID_INIT;
        }
        return versionId(data) + 1;
    }
```

Reply:
We can do that too. Then, we don't need to call `versionId(data)` since we already did the null check.

```
    private long getNextVersionId(String serviceUnit) {
        var data = tableview.get(serviceUnit);
        return getNextVersionId(data);
    }

    private long getNextVersionId(ServiceUnitStateData data) {
        return data == null ? VERSION_ID_INIT : data.getVersionId() + 1;
    }
```
Updated the code.

Nevertheless, if this is the first version(child bundle creation), we can pass the VERSION_ID_INIT without calling getNextVersionId(null). It looks redundant to call getNextVersionId(null) to get VERSION_ID_INIT.

`ServiceUnitStateData next = new ServiceUnitStateData(Owned, data.broker(), VERSION_ID_INIT);`
or 
`ServiceUnitStateData next = new ServiceUnitStateData(Owned, data.broker(), getNextVersionId(null));`